component/otelcol/config_filter.go

```diff
@@ -0,0 +1,215 @@
+		res["log_severity_texts"] = args.LogSeverityTexts
+	}
+
+	if args.LogSeverityNumber != nil {
+		res["log_severity_number"] = args.LogSeverityNumber.Convert()
+	}
+
+	if len(args.MetricNames) > 0 {
+		res["metric_names"] = args.MetricNames
+	}
+
+	if subRes := convertAttributeSlice(args.Attributes); len(subRes) > 0 {
+		res["attributes"] = subRes
+	}
+
+	if subRes := convertAttributeSlice(args.Resources); len(subRes) > 0 {
+		res["resources"] = subRes
+	}
+
+	if subRes := convertInstrumentationLibrariesSlice(args.Libraries); len(subRes) > 0 {
+		res["libraries"] = subRes
+	}
+
+	if len(args.SpanKinds) > 0 {
+		res["span_kinds"] = args.SpanKinds
+	}
+
+	return res
+}
+
+// Return an empty slice if the input slice is empty.
+func convertAttributeSlice(attrs []Attribute) []interface{} {
+	attrArr := make([]interface{}, 0, len(attrs))
+	for _, attr := range attrs {
+		attrArr = append(attrArr, attr.Convert())
+	}
+	return attrArr
+}
+
+// Return an empty slice if the input slice is empty.
+func convertInstrumentationLibrariesSlice(libs []InstrumentationLibrary) []interface{} {
+	libsArr := make([]interface{}, 0, len(libs))
+	for _, lib := range libs {
+		libsArr = append(libsArr, lib.Convert())
+	}
+	return libsArr
+}
+
+type RegexpConfig struct {
+	// CacheEnabled determines whether match results are LRU cached to make subsequent matches faster.
+	// Cache size is unlimited unless CacheMaxNumEntries is also specified.
+	CacheEnabled bool `river:"cacheenabled,attr,optional"`
+	// CacheMaxNumEntries is the max number of entries of the LRU cache that stores match results.
+	// CacheMaxNumEntries is ignored if CacheEnabled is false.
+	CacheMaxNumEntries int `river:"cachemaxnumentries,attr,optional"`
+}
+
+func (args RegexpConfig) Convert() map[string]interface{} {
+	return map[string]interface{}{
+		"cacheenabled":       args.CacheEnabled,
+		"cachemaxnumentries": args.CacheMaxNumEntries,
+	}
+}
+
+// Attribute specifies the attribute key and optional value to match against.
+type Attribute struct {
+	// Key specifies the attribute key.
+	Key string `river:"key,attr"`
+
+	// Values specifies the value to match against.
+	// If it is not set, any value will match.
+	Value interface{} `river:"value,attr,optional"`
+}
+
+func (args Attribute) Convert() map[string]interface{} {
+	return map[string]interface{}{
+		"key":   args.Key,
+		"value": args.Value,
+	}
+}
+
+// InstrumentationLibrary specifies the instrumentation library and optional version to match against.
+type InstrumentationLibrary struct {
+	Name string `river:"name,attr"`
+	// version match
+	//  expected actual  match
+	//  nil      <blank> yes
+	//  nil      1       yes
+	//  <blank>  <blank> yes
+	//  <blank>  1       no
+	//  1        <blank> no
+	//  1        1       yes
+	Version *string `river:"version,attr"`
+}
+
+func (args InstrumentationLibrary) Convert() map[string]interface{} {
+	return map[string]interface{}{
+		"name":    args.Name,
+		"version": strings.Clone(*args.Version),
```

Comment:
I think this will panic if `version = null` is configured in the River config, can we add a if statement here for safety?